tests/integration/test_blocks.py

```diff
@@ -0,0 +1,169 @@
+"""
+Integration tests for OCB.
+
+We use xvfb to perform the tests without opening any windows.
+We use pyautogui to move the mouse and interact with the application.
+
+To pass the tests on windows, you need to not move the mouse.
+Use this if you need to understand why a test fails.
+
+To pass the tests on linux, you just need to install xvfb and it's dependencies.
+On linux, no windows are opened to the user during the test.
+To understand why a test fails, pass the flag "--no-xvfb" and use your own X server
+to see the test running live.
+"""
+
+# Imports needed for testing
+import time, threading, queue, os, sys
+import pytest
+from pytest_mock import MockerFixture
+import pytest_check as check
+import pyautogui
+ 
+# Packages tested
+from opencodeblocks.graphics.blocks.codeblock import OCBCodeBlock
+from opencodeblocks.graphics.socket import OCBSocket
+from opencodeblocks.graphics.window import OCBWindow
+from opencodeblocks.graphics.widget import OCBWidget
+
+from qtpy.QtWidgets import QApplication
+from PyQt5.QtWidgets import QWidget
+from PyQt5.QtGui import QFocusEvent, QMouseEvent
+from PyQt5.QtCore import QCoreApplication, QEvent, Qt, QPointF, QPoint
+from PyQt5 import QtTest
+
+def test_window_opening(qtbot):
+    """ The OCBWindow should open and close correctly """
+    wnd = OCBWindow()
+    wnd.close()
+
+def test_running_python(qtbot):
+    """ The blocks should run arbitrary python when unfocused """
+    wnd = OCBWindow()
+    
+    EXPRESSION = "3 + 5 * 2"
+    SOURCE_TEST = \
+        '''
+            print(%s)
+        ''' % EXPRESSION
+    expected_result = str(eval(EXPRESSION))
+
+    # Let's add a block with the source to the window !
+    ocb_widget = OCBWidget()
+    test_block = OCBCodeBlock(title="Testing block", source=SOURCE_TEST)
+    ocb_widget.scene.addItem(test_block)
+    wnd.mdiArea.addSubWindow(ocb_widget)
+
+    # Let's run the block !
+    pyeditor = test_block.source_editor.widget()
+    # pyeditor.setModified(True)
+    # test_block._source = ""
+    QApplication.processEvents()
+    QtTest.QTest.mouseClick(pyeditor,Qt.MouseButton.LeftButton)
+    QApplication.processEvents()
+    QtTest.QTest.keyPress(pyeditor," ")
+    QApplication.processEvents()
+    
+    # Click outside the block to lose focus of the previous block.
+    # This will need to be changed by the click to the run button.
+    QtTest.QTest.mouseClick(ocb_widget,Qt.MouseButton.LeftButton)
+    QApplication.processEvents()
+
+    # When the execution becomes non-blocking for the UI, a refactor will be needed here.
+    result = test_block.stdout.strip()
+    
+    check.equal(expected_result,result)
+    wnd.close()
```

Comment:
This should be added in kernel branch only, it fails on master

Reply:
Is commenting it ok ?

Reply:
Yeah sure